pkg/network/ebpf/c/protocols/http2/decoding.h

```diff
@@ -473,10 +493,12 @@ static __always_inline bool get_first_frame(struct __sk_buff *skb, skb_info_t *s
     return false;
 }
 
-static __always_inline __u8 find_relevant_headers(struct __sk_buff *skb, skb_info_t *skb_info, http2_frame_with_offset *frames_array, __u8 original_index) {
+static __always_inline __u8 find_relevant_frames(struct __sk_buff *skb, skb_info_t *skb_info, http2_frame_with_offset *frames_array, __u8 original_index) {
     bool is_headers_or_rst_frame, is_data_end_of_stream;
     __u8 interesting_frame_index = 0;
     struct http2_frame current_frame = {};
+
+    // We may have found a relevant frame already in http2_handle_first_frame, so we need to adjust the index accordingly
```

Comment:
So, a reasonable reader would wonder why didn't we just do `__u8 interesting_frame_index = original_index`
And the reasonable reader won't find the answer

so I'd add a line that explains we're doing that, in that way to please the verifier